appinventor/components/src/com/google/appinventor/components/runtime/YandexTranslate.java

```diff
@@ -139,7 +148,7 @@ private void performRequest(String languageToTranslateTo, String textToTranslate
       throws IOException, JSONException {
 
     final String finalURL = YANDEX_TRANSLATE_SERVICE_URL +
-        this.yandexKey +
+      ((this.userYandexKey.equals("DEFAULT"))?this.yandexKey:this.userYandexKey) +
```

Comment:
See previous comment on equals and potential (although unlikely) NPE.

Reply:
Converted to TextUtils.equal which can accept a null pointer without faulting